File: fixtures/java/src/com/acme/order/OrderService.java
package com.acme.order;

import com.megacorp.ledger.LedgerClient;

/**
 * Billing and archival. Both {@code archive} overloads take a parameter whose
 * simple name is {@code List}; only their packages differ.
 */
public class OrderService {

    private final LedgerClient ledger;

    public OrderService(LedgerClient ledger) {
        this.ledger = ledger;
    }

    public Invoice bill(Order order) {
        Invoice invoice = new Invoice(order.reference());
        ledger.post(invoice);
        return invoice;
    }

    public void archive(java.util.List<Order> orders) {
        for (Order order : orders) {
            ledger.archive(order.reference());
        }
    }

    public void archive(com.acme.order.legacy.List orders) {
        ledger.archive(orders.head());
    }
}

File: fixtures/csharp/src/Acme/Order/OrderService.cs
using MegaCorp.Ledger;

namespace Acme.Order;

/// <summary>
/// Billing and archival. Both Archive overloads take a parameter whose simple
/// name is `List`; only their namespaces differ — and one is generic.
/// </summary>
public class OrderService
{
    private readonly LedgerClient ledger;

    private readonly Repository plain = new();

    private readonly Repository<Order> typed = new();

    public OrderService(LedgerClient ledger)
    {
        this.ledger = ledger;
    }

    public Invoice Bill(Order order)
    {
        Invoice invoice = new Invoice(order.Reference);
        ledger.Post(invoice);
        return invoice;
    }

    public void Archive(List<Order> orders)
    {
        foreach (Order order in orders)
        {
            ledger.Archive(order.Reference);
        }
    }

    public void Archive(Legacy.List orders)
    {
        ledger.Archive(orders.Head);
    }
}
